src/main/java/com/smartsheet/api/AttachmentVersioningResources.java

```diff
@@ -0,0 +1,93 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * %[license]
+ */
+import com.smartsheet.api.models.Attachment;
+import com.smartsheet.api.models.DataWrapper;
+import com.smartsheet.api.models.PaginationParameters;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.util.List;
+
+/**
+ * <p>This interface provides methods to access AttachmentVersioning resources.</p>
+ *
+ * <p>Thread Safety: Implementation of this interface must be thread safe.</p>
+ */
+public interface AttachmentVersioningResources {
+
+    /**
+     * <p>Deletes an attachment, including all of its versions.</p>
+     *
+     * <p>It mirrors to the following Smartsheet REST API method:<br />
+     * DELETE /sheets/{sheetId}/attachments/{attachmentId}/versions</p>
+     *
+     * @param sheetId the sheet id
+     * @param attachentId the attachment id
+     * @throws IllegalArgumentException if any argument is null or empty string
+     * @throws InvalidRequestException if there is any problem with the REST API request
+     * @throws AuthorizationException if there is any problem with  the REST API authorization (access token)
+     * @throws ResourceNotFoundException if the resource cannot be found
+     * @throws ServiceUnavailableException if the REST API service is not available (possibly due to rate limiting)
+     * @throws SmartsheetException if there is any other error during the operation
+     */
+    public void deleteAllVersions(long sheetId, long attachentId) throws SmartsheetException;
+
+    /**
+     * <p>Get all versions of an attachment.</p>
+     *
+     * <p>It mirrors to the following Smartsheet REST API method:<br />
+     * GET /sheets/{sheetId}/attachments/{attachmentId}/versions</p>
+     *
+     * @param sheetId the id
+     * @param attachmentId the attachment id
+     * @param parameters the pagination paramaters
+     * @return the attachment (note that if there is no such resource, this method will throw ResourceNotFoundException
+     * rather than returning null).
+     * @throws IllegalArgumentException if any argument is null or empty string
+     * @throws InvalidRequestException if there is any problem with the REST API request
+     * @throws AuthorizationException if there is any problem with  the REST API authorization (access token)
+     * @throws ResourceNotFoundException if the resource cannot be found
+     * @throws ServiceUnavailableException if the REST API service is not available (possibly due to rate limiting)
+     * @throws SmartsheetException if there is any other error during the operation
+     */
+    public DataWrapper<Attachment> listAllVersions(long sheetId, long attachmentId, PaginationParameters parameters) throws SmartsheetException;
```

Comment:
DataWrapper class would be easier to understand if it were labelled something like "PagedResult<T>", since it is used exclusively with the paged GET calls. It took me a bit to figure out what a "DataWrapper"'s purpose was.